### PR TITLE
Configure as extension only for watch extensions

### LINKED ISSUE #3153: WatchKit Extension Pods Not Setting Extension API CheckBox
I am currently developing a WatchKit Kit app using some cocoa pods. The fix for #2980 works to set the "Allow app extension API only" checkbox for a today widget in the same application, but it does not set the check box for the WatchKit Extension.

My guess is the detection used to detect other extensions is slightly different for a watch kit extension.

### PR DESCRIPTION
**Pull request by neonichu**
Fixes #3153

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,10 @@ To install release candidates run `[sudo] gem install cocoapods --pre`
 
 ##### Enhancements
 
+* Set the APPLICATION_EXTENSION_API_ONLY build setting if integrating with a watch extension target.
```
Comment (segiddins) [inline CHANGELOG.md]: Use ` around `APPLICATION_EXTENSION_API_ONLY` ?

Comment (segiddins) [inline CHANGELOG.md]: Also, new changelog entries typically go to the bottom?

**Comment by segiddins** [comment]
:+1:

File: CHANGELOG.md (hunk 1 above)
Comment (neonichu) [inline CHANGELOG.md]: Do they? I have always been putting them at the top :)

**Pull request merged**